conanfile.py

```diff
@@ -135,6 +143,176 @@ def _python_installs(self):
+                    self.output.info(f"Retrieving license for {package}")
+                    response = requests.get(sources_url)
+                    response.raise_for_status()
+
+                    with tempfile.TemporaryDirectory() as temp_dir:
+                        sources_path = os.path.join(temp_dir, "sources.tar.gz")
+                        with open(sources_path, 'wb') as sources_file:
+                            sources_file.write(response.content)
+
+                        with tarfile.open(sources_path, 'r:gz') as sources_archive:
+                            license_file = "LICENSE"
+
+                            for source_file in sources_archive.getnames():
+                                if Path(source_file).name == license_file:
+                                    sources_archive.extract(source_file, temp_dir)
+
+                                    license_file_path = os.path.join(temp_dir, source_file)
+                                    with open(license_file_path, 'r') as file:
+                                        dependency_description["license_full"] = file.read()
+
+        for source_url, check_source in [("source", False),
+                                         ("Source", False),
+                                         ("Source Code", False),
+                                         ("Repository", False),
+                                         ("Code", False),
+                                         ("homepage", True),
+                                         ("Homepage", True)]:
+            try:
+                url = data["info"]["project_urls"][source_url]
+                if check_source and not self._is_repository_url(url):
+                    # That will not work for ALL open-source projects, but should already get a large majority of them
+                    continue
+                dependency_description["sources_url"] = url
+                break
+            except KeyError:
+                pass
+
+        if dependency_description["license"] is not None and len(dependency_description["license"]) > 32:
+            # Some packages have their full license in this field
+            dependency_description["license_full"] = dependency_description["license"]
+            dependency_description["license"] = data["info"]["name"]
+
+        dependencies[data["info"]["name"]] = dependency_description
+
+    @staticmethod
+    def _get_license_from_repository(sources_url, version, license_file_name = None):
+        git_url = sources_url
+        if git_url.endswith('/'):
+            git_url = git_url[:-1]
+        if not git_url.endswith(".git"):
+            git_url = f"{git_url}.git"
+        git_url = git_url.replace("/cgit/", "/")
+
+        tags = [f"v{version}", version]
+        files = ["LICENSE", "LICENSE.txt", "LICENSE.md", "COPYRIGHT", "COPYING", "COPYING.LIB"] if license_file_name is None else [license_file_name]
```

Comment:
Could this also be (in) a folder sometimes?

(Also spelling, since we saw that LICENCE is technically correct 😛)

Reply:
Again, this is based on what I found on the repositories of our actual dependencies. When they do a folder, that usually means they have multiple licences so I don't know which one to pick. This is a best-effort.